engineapi/backup_monitor.go

```diff
@@ -109,6 +123,68 @@ func NewBackupMonitor(logger logrus.FieldLogger,
 	return m, nil
 }
 
+// getVolumeRecurringJobInfos get recurring jobs in the volume labels and recurring jobs of groups in the volume labels
+func (m *BackupMonitor) getVolumeRecurringJobInfos(ds *datastore.DataStore, volume *longhorn.Volume) (string, error) {
+	allRecurringJobs, err := ds.ListRecurringJobsRO()
+	if err != nil {
+		m.logger.WithError(err).Warn("Failed to list all recurring jobs")
+		return "", err
+	}
+
+	volumeRecurringJobInfos := make(map[string]longhorn.VolumeRecurringJobInfo)
+	// distinguish volume labels between job and group.
+	volumeSelectedRecurringJobs := datastore.MarshalLabelToVolumeRecurringJob(volume.Labels)
+	for recurringJobName, recurringJob := range allRecurringJobs {
+		for jobName, job := range volumeSelectedRecurringJobs {
+			if job.IsGroup {
+				// add the recurring job from group or update its groups informtaion
+				if volumeRecurringJobInfos, err = addVolumeRecurringJobInfosFromGroup(recurringJobName, jobName, &recurringJob.Spec, volumeRecurringJobInfos); err != nil {
+					m.logger.WithError(err).WithFields(logrus.Fields{"recurringjob": recurringJobName, "group": jobName}).Warn("Failed to add the recurring job")
```

Comment:
nit: Suggest reviewing `when we should print logs`? the basic pattern is as below (no need to do it right now, but we should review this in general later to make our logging concise)

- If the error will be returned to the caller, it should not be printed out except some extra logging
- If the error will not be returned to caller, need to print the log directly